### DISCUSSION TITLE
Is it possible to make inline assembly on Apple M1 machine?

### DISCUSSION THREAD

**Discussion by petaqb** [vlang/v#19219] [Questions and Answers]
It seems to me that M1 is not implemented...
Thanks

**Comment by JalonSolov** [comment]
```v
asm arm64 {
    // arm64 asm goes here
}
```

**Reply by petaqb** [reply]
Thank you! But is there any example how it works?

**Reply by JalonSolov** [reply]
`examples/asm.v` has an example using amd64 asm.

**Reply by petaqb** [reply]
I see. Unfortunately I got an error for arm64 (on macOs):

asm.v:13:1: error: invalid token in assembly block
   11 |     asm arm64 {
   12 |   
   13 |     MOV x0, #1      // 1 = StdOut
      | ~~~~~~~~~~~~~~~~~~~~~~

The source is:

	mystr := "hello!!!"

	asm arm64 {
  
    MOV x0, #1      // 1 = StdOut
    ADR x1, mystr  // string to print
    MOV x2, #8         // length of our string
    MOV x16, #4     // Unix write system call
    SVC #0x80       // Call kernel to output the string
		
	
	}

**Reply by JalonSolov** [reply]
Couple of things wrong (at least for V)
```v
mystr := "hello!!!"

asm arm64 {
	MOV x0, #1         // 1 = StdOut
	ADR x1, mystr.str  // string to print
	MOV x2, mystr.len  // length of our string
	MOV x16, #4        // Unix write system call
	SVC #0x80          // Call kernel to output the string
}
```
However, the thing it's complaining about is the `#` in the first statement.  I don't know what should go there.  You might want to look up the docs for gcc inline asm, etc., as the C compilers are the ones that handle this.
